tests/validator/test_validation_graph.py

```diff
@@ -178,10 +249,226 @@ def test_ExpectationValidationGraph_get_exception_info(
+    basic_datasource,
+):
+    df = pd.DataFrame({"a": [1, 5, 22, 3, 5, 10], "b": [1, 2, 3, 4, 5, None]})
+
+    batch = basic_datasource.get_single_batch_from_batch_request(
+        RuntimeBatchRequest(
+            **{
+                "datasource_name": "my_datasource",
+                "data_connector_name": "test_runtime_data_connector",
+                "data_asset_name": "IN_MEMORY_DATA_ASSET",
+                "runtime_parameters": {
+                    "batch_data": df,
+                },
+                "batch_identifiers": {
+                    "pipeline_stage_name": 0,
+                    "airflow_run_id": 0,
+                    "custom_key_0": 0,
+                },
+            }
+        )
+    )
+
+    expectation_configuration = ExpectationConfiguration(
+        expectation_type="expect_column_max_to_be_between",
+        kwargs={"column": "not_in_table", "min_value": 1, "max_value": 29},
+    )
+
+    runtime_configuration = {
+        "catch_exceptions": True,
+        "result_format": {"result_format": "BASIC"},
+    }
+
+    execution_engine = PandasExecutionEngine()
+
+    # noinspection PyUnusedLocal
+    validator = Validator(execution_engine=execution_engine, batches=[batch])
+
+    validation_dependencies = ExpectColumnMaxToBeBetween().get_validation_dependencies(
+        expectation_configuration, execution_engine, runtime_configuration
+    )["metrics"]
+
+    graph = ValidationGraph(execution_engine=execution_engine)
+
+    for metric_configuration in validation_dependencies.values():
+        graph.build_metric_dependency_graph(
+            metric_configuration=metric_configuration,
+            runtime_configuration=runtime_configuration,
+        )
+
+    metrics: Dict[Tuple[str, str, str], Any] = {}
+    aborted_metrics_info: Dict[
+        Tuple[str, str, str],
+        Dict[str, Union[MetricConfiguration, Set[ExceptionInfo], int]],
+    ] = graph.resolve_validation_graph(
+        metrics=metrics,
+        runtime_configuration=runtime_configuration,
+    )
+
+    assert len(aborted_metrics_info) == 1
+
+    aborted_metric_info_item = list(aborted_metrics_info.values())[0]
+    assert aborted_metric_info_item["num_failures"] == MAX_METRIC_COMPUTATION_RETRIES
+
+    assert len(aborted_metric_info_item["exception_info"]) == 1
+
+    exception_info = next(iter(aborted_metric_info_item["exception_info"]))
+    assert (
+        exception_info["exception_message"]
+        == 'Error: The column "not_in_table" in BatchData does not exist.'
+    )
+
+
+@pytest.mark.unit
+def test_progress_bar_config_enabled():
+    class DummyMetricConfiguration:
+        pass
+
+    class DummyExecutionEngine:
+        pass
+
+    dummy_metric_configuration = cast(MetricConfiguration, DummyMetricConfiguration)
+    dummy_execution_engine = cast(ExecutionEngine, DummyExecutionEngine)
+
+    # ValidationGraph is a complex object that requires len > 3 to not trigger tqdm
+    with mock.patch(
```

Comment:
this series of mocks is duplicated between this method and the next; consider factoring into its own method or fixture

Reply:
@donaldheppner @Kilo59 was explicitly against doing this; we should discuss together.  Thanks.

Reply:
I'm fine with it not being a fixture, but a well named method here is then the way to go. Duplication in tests is still duplication, and there is value in knowing that both tests use the same setup